### DISCUSSION TITLE
TypeError: __jsx is not a function

### DISCUSSION THREAD

**Discussion by Onibenjo** [vercel/next.js#17329] [Help] [answered]
# Bug report

## Describe the bug
using custom _app.js
I keep getting 
`Server Error
TypeError: __jsx is not a function` at 
```javascript 
const MyApp = ({ Component, pageProps, store }) => <Component {...pageProps} />
```

## To Reproduce
1. git clone https://github.com/adesiyanbs/preeson-2.git
2. cd preeson-2
3. yarn
4. yarn dev

## Expected behavior
Compile successful

## Screenshots

![](https://github.com/Onibenjo/errors/blob/master/__jsx%20nt%20func.png?raw=true)

## System information

- OS: [e.g. macOS, Windows]
- Browser (if applies) [e.g. chrome, safari]
- Version of Next.js: [e.g. 6.0.2]
- Version of Node.js: [e.g. 10.10.0]

**Comment by gabrielalmeida** [comment, marked as answer]
Hi!

There are several errors on your `_app.js` preventing your page to render.

* `import React from 'react-dom'`

`React` doesn't need to be [imported if you're only willing to use JSX](https://github.com/vercel/next.js/issues/12964#issuecomment-629642157), if you're willing to use something available under `React`, you need to import it by doing `import React from 'react'`.

* Wrong usage of `getInitialProps`

I don't get what you're [trying to achieve here](https://github.com/adesiyanbs/preeson-2/blob/16f4e7f5e93d5e8e0f185eec5d03fd9b657c5fe8/pages/_app.js#L14-L17):

```js
  static async getInitialProps({Component, ctx}) {
    const pageProps = Component.getInitialProps ? await Component.getInitialProps(ctx) : {};
     return {pageProps: pageProps};
    }
```

The [only argument `getInitialProps` receives is the `ctx` itself.](https://nextjs.org/docs/api-reference/data-fetching/getInitialProps#context-object) So this line is currently receiving `undefined` for these destructured values:

```js 
static async getInitialProps({Component, ctx}) {
```

Besides that, it seems you're trying to replace `pageProps`, which is also unexpected.

It seems you're trying to have blocking data requirements for every single page in your application, [which should probably be avoided](https://nextjs.org/docs/advanced-features/custom-app):

```
// Only uncomment this method if you have blocking data requirements for
// every single page in your application. This disables the ability to
// perform automatic static optimization, causing every page in your app to
// be server-side rendered.
```

But if that's really what you're trying to do, replace your `getInitialProps` with something along the lines:

```js
  static async getInitialProps(ctx) {
    const appProps = await App.getInitialProps(ctx);
    return { ...appProps };
  }
```

Your usage of [next-redux-wrapper](https://github.com/kirill-konshin/next-redux-wrapper) is also wrong. Note the correct imports and wrapper stated on docs:

```js
import { createWrapper }from "next-redux-wrapper";
const store = createStore(cartReducer);
const makestore = () => store;
const wrapper = createWrapper(makestore, {debug: true})
export default wrapper.withRedux(MyApp)
```

Note that at the end of the [usage `next-redux-wrapper` section](https://github.com/kirill-konshin/next-redux-wrapper#usage) they state:

> ⚠️ Next.js provides generic getInitialProps when using class MyApp extends App which will be picked up by wrapper, so you **must not extend App** as you'll be opted out of Automatic Static Optimization: https://err.sh/next.js/opt-out-auto-static-optimization. Just export a regular Functional Component as in the example above.

So you'll probably need to rewrite this as a Functional Component to use that lib.

**Reply by Onibenjo** [reply]
Thank you so much. This clarified things

Thanks

**Reply by Onibenjo** [reply]
and it worked
